workflow/controller/operator_test.go

```diff
@@ -6005,3 +6005,200 @@ func TestHasOutputResultRef(t *testing.T) {
 	assert.True(t, hasOutputResultRef("generate-random", &wf.Spec.Templates[0]))
 	assert.True(t, hasOutputResultRef("generate-random-1", &wf.Spec.Templates[0]))
 }
+
+const stepsFailFast = `
```

Comment:
I've already ensured this is minimal :)

This one is particularly longer because it tests nested boundary behavior: `failFast` should only affect a single depth of boundary and not be affected by children boundaries (such as a `retryStrategy`)